examples/chip-tool/commands/clusters/WriteAttributeCommand.h

```diff
@@ -105,6 +110,38 @@ class WriteAttribute : public ModelCommand, public chip::app::WriteClient::Callb
         return mWriteClient->SendWriteRequest(device->GetSecureSession().Value());
     }
 
+    template <class T>
+    CHIP_ERROR SendGroupCommand(chip::GroupId groupId, chip::FabricIndex fabricIndex, chip::NodeId senderNodeId,
+                                chip::ClusterId clusterId, chip::AttributeId attributeId, const T & value)
+    {
+
+        chip::app::AttributePathParams attributePathParams;
+        attributePathParams.mClusterId   = clusterId;
+        attributePathParams.mAttributeId = attributeId;
+
+        chip::Messaging::ExchangeManager * exchangeManager = chip::app::InteractionModelEngine::GetInstance()->GetExchangeManager();
+
+        ChipLogDetail(chipTool, "Sending Write Attribute to Group %u, on Fabric %x, for cluster %u with attributeId %u", groupId,
+                      fabricIndex, clusterId, attributeId);
+
+        auto writeClient = chip::Platform::MakeUnique<chip::app::WriteClient>(exchangeManager, this, mTimedInteractionTimeoutMs);
+        VerifyOrReturnError(writeClient != nullptr, CHIP_ERROR_NO_MEMORY);
+        ReturnErrorOnFailure(writeClient->EncodeAttribute(attributePathParams, value, mDataVersion));
+
+        chip::Optional<chip::SessionHandle> session =
+            exchangeManager->GetSessionManager()->CreateGroupSession(groupId, fabricIndex, senderNodeId);
+        if (!session.HasValue())
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
+        ReturnErrorOnFailure(writeClient->SendWriteRequest(session.Value()));
+
+        writeClient.release();
+        exchangeManager->GetSessionManager()->RemoveGroupSession(session.Value()->AsGroupSession());
```

Comment:
This is an odd ownership model - why can't the session automatically be cleaned up given that a) it's a group session and b) the number of holders of that session go to 0?

Reply:
That cannot be done automatically given the current timeframe. An issue is open for 1.X however.